test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/FormActionTagHelperTest.cs

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers
+{
+    public class FormActionTagHelperTest
+    {
+        [Fact]
+        public async Task ProcessAsync_GeneratesExpectedOutput()
```

Comment:
:see_no_evil: Since our `UrlHelper` tests are extensive, the most important thing to confirm about the `FormActionTagHelper` is that it calls the `IUrlHelper` correctly. The unit tests below cover that.

This is a quick integration test that confirms everything is working end-to-end. All a functional test would add is checks of the `[HtmlTargetElement]` attributes and underlying Razor bits.